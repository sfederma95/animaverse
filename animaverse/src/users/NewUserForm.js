import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';
import Alert from '../Alert'
import './register.css';
import sleepCat from './sleepcat.svg';

function NewUserForm({register}){
    const history = useHistory();
    const [formData,setFormData]=useState({
        username: "",
        password: "",
        email: "",
        avatar: "",
    })
    const [formErrors,setFormErrors] = useState([])
    async function handleSubmit(e){
        e.preventDefault();
        let res = await register(formData);
        if (res.errors) {
            setFormErrors(res.errors)
        } else{
            history.push(`/users/${res.user.usr_id}`)
        }
    }
    function handleChange(e){
        const {name,value} = e.target;
        setFormData(l=>({...l,[name]:value}))
    }
    return(
        <div id='reg-form'>
            <h1 id='tagline'>Sign up to start playing with Pixeloos</h1>
            <form id='form-el' onSubmit={handleSubmit}>
                <p id='register-p'>Register:</p>
                <div className='inputs'>
                    <img className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='username'>Username: </label>
                    <input className='input-el' id='username' name='username' value={formData.username} onChange={handleChange} type='text' required/>
                </div>
                <div className='inputs'>
                    <img className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='password'>Password: </label>
                    <input className='input-el' id='password' name='password' value={formData.password} onChange={handleChange} type='text' required/>
                </div>
                <div className='inputs'>
                    <img className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='email'>Email: </label>
                    <input className='input-el' id='email' name='email' value={formData.email} onChange={handleChange} type='text' required/>
                </div>
                <div className='inputs'>
                    <img className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='avatar'>Avatar URL: </label>
                    <input className='input-el' id='avatar' name='avatar' value={formData.avatar} onChange={handleChange} type='text'/>
                </div>
                {formErrors.length ? <Alert messages={formErrors}/> : null}
                <button id='reg-btn'>Register</button>
            </form>
        </div>
    )
}

export default NewUserForm;