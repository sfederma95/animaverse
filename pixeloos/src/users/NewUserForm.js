import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';
import Alert from '../Alert'
import './register.css';
import sleepCat from './sleepcat.svg';
import {hideCat, showCat} from './forms'

function NewUserForm({register}){
    const history = useHistory();
    const [formData,setFormData]=useState({
        username: "",
        password: "",
        email: "",
        avatar: "",
    })
    const [formErrors,setFormErrors] = useState([])
    async function handleSubmit(e){
        e.preventDefault();
        let res = await register(formData);
        if (res.errors) {
            setFormErrors(res.errors)
        } else{
            history.push(`/users/${res.user.usr_id}`)
        }
    }
    function handleChange(e){
        const {name,value} = e.target;
        setFormData(l=>({...l,[name]:value}))
    }
    
    return(
        <div id='reg-form'>
            <h1 id='tagline'>Sign up to start playing with Pixeloos</h1>
            <form id='form-el' onSubmit={handleSubmit}>
                <p id='register-p'>Register:</p>
                <div className='inputs'>
                    <img alt='sleep cat' className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='username'>Username: </label>
                    <input onFocus = {showCat} onBlur = {hideCat} className='input-el good-input' id='username' name='username' value={formData.username} onChange={handleChange} type='text'/>
                </div>
                <div className='inputs'>
                    <img alt='sleep cat' className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='password'>Password: </label>
                    <input onFocus = {showCat} onBlur = {hideCat} className='input-el good-input' id='password' name='password' value={formData.password} onChange={handleChange} type='text' />
                </div>
                <div className='inputs'>
                    <img alt='sleep cat' className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='email'>Email: </label>
                    <input onFocus = {showCat} onBlur = {hideCat} className='input-el good-input' id='email' name='email' value={formData.email} onChange={handleChange} type='text' />
                </div>
                <div className='inputs'>
                    <img alt='sleep cat' className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='avatar'>Avatar URL: </label>
                    <input onFocus = {showCat} onBlur = {hideCat} className='input-el good-input' id='avatar' name='avatar' value={formData.avatar} onChange={handleChange} type='text'/>
                </div>
                {formErrors.length ? <Alert messages={formErrors}/> : null}
                <button id='reg-btn'>Register</button>
            </form>
        </div>
    )
}

export default NewUserForm;