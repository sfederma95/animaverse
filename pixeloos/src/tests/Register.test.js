import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import NewUserForm from '../users/NewUserForm';
import {UserProvider} from './testSetup';

it ('renders registration form component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <NewUserForm />
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <NewUserForm/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})