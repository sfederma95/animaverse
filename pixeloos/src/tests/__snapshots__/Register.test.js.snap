// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="reg-form"
  >
    <h1
      id="tagline"
    >
      Sign up to start playing with Pixeloos
    </h1>
    <form
      id="form-el"
    >
      <p
        id="register-p"
      >
        Register:
      </p>
      <div
        class="inputs"
      >
        <img
          alt="sleep cat"
          class="sleep-cat"
          src="sleepcat.svg"
        />
        <label
          class="labels"
          for="username"
        >
          Username: 
        </label>
        <input
          class="input-el good-input"
          id="username"
          name="username"
          type="text"
          value=""
        />
      </div>
      <div
        class="inputs"
      >
        <img
          alt="sleep cat"
          class="sleep-cat"
          src="sleepcat.svg"
        />
        <label
          class="labels"
          for="password"
        >
          Password: 
        </label>
        <input
          class="input-el good-input"
          id="password"
          name="password"
          type="text"
          value=""
        />
      </div>
      <div
        class="inputs"
      >
        <img
          alt="sleep cat"
          class="sleep-cat"
          src="sleepcat.svg"
        />
        <label
          class="labels"
          for="email"
        >
          Email: 
        </label>
        <input
          class="input-el good-input"
          id="email"
          name="email"
          type="text"
          value=""
        />
      </div>
      <div
        class="inputs"
      >
        <img
          alt="sleep cat"
          class="sleep-cat"
          src="sleepcat.svg"
        />
        <label
          class="labels"
          for="avatar"
        >
          Avatar URL: 
        </label>
        <input
          class="input-el good-input"
          id="avatar"
          name="avatar"
          type="text"
          value=""
        />
      </div>
      <button
        id="reg-btn"
      >
        Register
      </button>
    </form>
  </div>
</DocumentFragment>
`;
